rfcs/01924_tls_attestation_ca.md

```diff
@@ -0,0 +1,169 @@
+# RFC#01924: Remote Attestation via TLS (Certificate Authority)
+
+## Objective
+
+Remotely attest applications in Trusted Execution Environments (TEEs) over the
+TLS protocol.
+
+## Background
+
+A lot of existing applications that would benefit from the remote attestation
+use **TLS protocol** to create secure channels with their clients. But
+integrating the remote attestation might bring up unnecessary complications in a
+form of completely rewriting the application connection logic.
+
+Thus, instead of refactoring existing applications and implementing a new
+protocol, we propose to integrate remote attestation into TLS by
+cryptographically binding **TLS certificates** to the **TEE firmware**.
+
+## Overview
+
+In order to allow applications to use TLS for remote attestation, we propose to
+use a **Certificate Authority (CA)** that generates signed TLS certificates for
+TEE applications. The main idea is that instead of attesting TEE applications
+themselves, clients attest the CA, which attests TEE applications on behalf of
```

Comment:
nit: "attest" is used inaccurately throughout. Attesting is providing proof, not validating the proof. The server attests to the software it is running and the client checks or validates the attestation.

Reply:
Using attestation to establish trust in the CA implies that the CA itself is running inside some sort of TEE environment (or perhaps using a TPM). This may not always be necessary (for example if the entity operating the client is also operating the CA, in which case they already trust it).